include/verilogAST.hpp

```diff
@@ -222,31 +222,19 @@ class Slice : public Expression {
 };
 
 class Index : public Expression {
-  std::variant<std::unique_ptr<Identifier>, std::unique_ptr<Attribute>,
-               std::unique_ptr<Slice>>
-  clone_index_value() const {
-    return std::visit(
-        [](auto&& value) -> std::variant<std::unique_ptr<Identifier>,
-                                         std::unique_ptr<Attribute>,
-                                         std::unique_ptr<Slice>> {
-          return value->clone();
-        },
-        this->value);
-  }
-
  protected:
   virtual Index* clone_impl() const override {
     return new Index(this->clone_index_value(), this->index->clone());
   };
 
  public:
   std::variant<std::unique_ptr<Identifier>, std::unique_ptr<Attribute>,
-               std::unique_ptr<Slice>>
+               std::unique_ptr<Slice>, std::unique_ptr<Index>>
       value;
   std::unique_ptr<Expression> index;
 
   Index(std::variant<std::unique_ptr<Identifier>, std::unique_ptr<Attribute>,
-                     std::unique_ptr<Slice>>
+                     std::unique_ptr<Slice>, std::unique_ptr<Index>>
```

Comment:
why is this necessary for this change?

Reply:
For multi-dimensional arrays, we need to nest index nodes (e.g. `x[0][1]`), so an Index node needs to be able to have an Index as it's value